app/src/main/java/com/example/topmovies/adapter/MoviesAdapter.kt

```diff
@@ -55,26 +56,47 @@ class MovieAdapter(private val onItemClickListener: (String) -> Unit) :
                 textviewItemLayoutRankNumber.text = movie.rank
                 textviewItemLayoutYearNumber.text = movie.year
                 textviewItemLayoutPreviousRankNumber.text = movie.rankUpDown
-                when (movie.rankUpDown.first()) {
-                    '+' -> textviewItemLayoutPreviousRankNumber.setTextColor(
-                        ContextCompat.getColor(itemView.context, R.color.text_rank_up)
-                    )
-                    '-' -> textviewItemLayoutPreviousRankNumber.setTextColor(
-                        ContextCompat.getColor(itemView.context, R.color.text_rank_down)
-                    )
-                }
                 circleAvatarViewItemLayoutMovieImage.setLabel(movie.title)
                 Glide.with(circleAvatarViewItemLayoutMovieImage)
                     .asBitmap()
                     .load(resizeImage(movie.imageUrl))
                     .into(avatarCustomTarget)
+                imageButtonItemFavoriteIcon.setImageResource(movieIsFavorite(movie.isFavorite))
+                setRankUpDownColor(movie.rankUpDown)
+                imageButtonItemFavoriteIcon.setOnClickListener {
+                    setOnClickListenerFavoriteButton(movie)
+
+                }
             }
-            itemView.setOnClickListener {
-                click(movie.id)
+            itemView.setOnClickListener { onItemClickListener(movie.id) }
+        }
+
+        private fun setRankUpDownColor(rankUpDown: String) {
+            binding.textviewItemLayoutPreviousRankNumber.setTextColor(
+                ContextCompat.getColor(itemView.context,
+                    when (rankUpDown.first()) {
+                        RANK_UP -> R.color.text_rank_up
+                        RANK_DOWN -> R.color.text_rank_down
+                        else -> R.color.text_color_fresh_ivy_green
+                    }
+                ))
+        }
+
+        private fun setOnClickListenerFavoriteButton(movie: Movie) {
+            if (!movie.isFavorite) {
+                binding.imageButtonItemFavoriteIcon.setImageResource(R.drawable.ic_filled_star)
+                movie.isFavorite = true
+            } else {
+                binding.imageButtonItemFavoriteIcon.setImageResource(R.drawable.ic_unfilled_star)
+                movie.isFavorite = false
             }
         }
 
-        private fun resizeImage(image: String) =
-            image.replaceAfter(REPLACE_AFTER, IMAGE_SIZE)
+        private fun movieIsFavorite(isFavorite: Boolean): Int {
```

Comment:
"is" should be the prefix of this naming.
Pay attention to how you are naming things❗❗❗❗❗❗
🙏🏽🙏🏽🙏🏽🙏🏽🙏🏽

Reply:
Oh well, you should rename it fully.
Please provide a corresponding naming.